tasks/run_cd4pe_job.rb

```diff
@@ -19,15 +19,52 @@ def docker_run_args
   @params["docker_run_args"]
 end
 
+def job_instance_id
+  @params["job_instance_id"]
+end
+
+def get_combined_exit_code(output)
+  job = output[:job]
+  after_job_success = output[:after_job_success]
+  after_job_failure = output[:after_job_failure]
+
+  exit_code_sum = job[:exit_code]
+  if (!after_job_success.nil?)
+    exit_code_sum = exit_code_sum + after_job_success[:exit_code]
+  end
+
+  if (!after_job_failure.nil?)
+    exit_code_sum = exit_code_sum + after_job_failure[:exit_code]
+  end
+
+  exit_code_sum == 0 ? 0 : 1
+end
+
 begin
+  @@logs = []
```

Comment:
Do we have to worry about volume of logging? In other words, is this a high traffic path?

Reply:
Nope! Luckily all the logs being added to this particular array are pushhed in one-by-one by our `run_cd4pe_job.rb` script, and the `cd4pe_job_helper.rb` helper